src/Collapse.js

```diff
@@ -0,0 +1,101 @@
+import React, { Component, PropTypes } from 'react';
+import classNames from 'classnames';
+import omit from 'lodash.omit';
+
+const SHOW = 'SHOW';
+const SHOWN = 'SHOWN';
+const HIDE = 'HIDE';
+const HIDDEN = 'HIDDEN';
+
+const propTypes = {
+  isOpen: PropTypes.bool,
+  className: PropTypes.node
+};
+
+const defaultProps = { isOpen: false };
+
+class Collapse extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      collapse: props.isOpen ? SHOWN : HIDDEN,
+    };
+    this.element = null;
+  }
+
+  componentWillReceiveProps(nextProps) {
+    const willOpen = nextProps.isOpen;
+    const collapse = this.state.collapse;
+
+    if (willOpen && collapse === HIDDEN) {
+      // will open
+      this.setState({ collapse: SHOW }, () => {
+        // start transition
+        this.element.style.height = `${this.getHeight()}px`;
+      });
+      this.transitionTag = setTimeout(() => {
+        this.setState({ collapse: SHOWN });
```

Comment:
I believe the bootstrap js removes the height inline style after it has been shown. This would likely prevent any issues when added content and then collapsing the content again.